10.0/s2i/bin/assemble

```diff
@@ -197,20 +276,14 @@ if [ -f "$LOCAL_SOURCE_DIR/pom.xml" ]; then
   popd &> /dev/null
 else
   echo "Copying binaries in source directory into $DEPLOY_DIR for later deployment..."
-  cp $LOCAL_SOURCE_DIR/*.war $DEPLOY_DIR >& /dev/null
-  chgrp -R 0 $DEPLOY_DIR
-  chmod -R g+rw $DEPLOY_DIR
+  copy_artifacts "$LOCAL_SOURCE_DIR" war ear rar jar
 fi
 
 if [ -d $LOCAL_SOURCE_DIR/target ]; then
-  cp $LOCAL_SOURCE_DIR/target/*.war $DEPLOY_DIR >& /dev/null
-  chgrp -R 0 $DEPLOY_DIR
-  chmod -R g+rw $DEPLOY_DIR
+  copy_artifacts "target" war ear rar jar
 fi
 if [ -d $LOCAL_SOURCE_DIR/deployments ]; then
-  cp $LOCAL_SOURCE_DIR/deployments/*.war $DEPLOY_DIR >& /dev/null
-  chgrp -R 0 $DEPLOY_DIR
-  chmod -R g+rw $DEPLOY_DIR
+  copy_artifacts "deployments" war ear rar jar
```

Comment:
hm, perhaps you are correct.

as long as the end result is that when you oc rsh into the pod, you can delete/overwrite the files, that's sufficient.